var/spack/repos/builtin/packages/py-pbr/package.py

```diff
@@ -34,3 +34,4 @@ class PyPbr(PythonPackage):
     version('1.8.1', 'c8f9285e1a4ca6f9654c529b158baa3a')
 
     depends_on('py-setuptools', type='build')
+    depends_on('py-enum34', type='build')
```

Comment:
That would look like:
```python 
depends_on('enum34', when='^python@:3.4.2')
```
However, I would highly recommend against using that. Concretization is currently very broken, and this exact syntax has caused me woes in the past. See `py-flake8` and dependencies where I had to comment this out to get it to work.